libparsers/tests/xml/dynamic_background.cpp

```diff
@@ -0,0 +1,152 @@
+#include "parsers/xml/dynamic_background.hpp"
+#include "utils.hpp"
+#include <doctest.h>
+#include <iostream>
+
+using namespace parsers::xml;
+using parsers::errors::error_t;
+using parsers::errors::errors_t;
+using namespace openbve2::datatypes;
+
+TEST_SUITE_BEGIN("libparsers - xml - dynamic_background");
+
+TEST_CASE("libparser - xml - dynamic_background - test single backgound") {
+	// clang-format off
+    std::string const test_single_background = 
+        "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
+        "<openBVE xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">"
+            "<Background>"
+                "<Time>00.00</Time>"
+                "<Mode>FadeIn</Mode>"
+                "<Repetitions>6</Repetitions>"
+                "<Texture>Cloudy.png</Texture>"
+            "</Background>"
+        "</openBVE>";
+	// clang-format on
+	parsers::errors::multi_error_t output_errors;
+	auto rel_file_func = [](std::string const& base, std::string const& relative) {
+		return base + "/" + relative;
+	};
+	auto const output = dynamic_background::parse("some_file.xml", test_single_background,
+	                                              output_errors, rel_file_func);
+
+	if (!output_errors.empty()) {
+		std::cout << output_errors;
+	}
+	REQUIRE(output_errors.empty());
+
+	REQUIRE(output.is<std::vector<dynamic_background::texture_background_info>>());
+	auto& background =
+	    output.get_unchecked<std::vector<dynamic_background::texture_background_info>>();
+	CHECK_EQ(background.size(), 1);
```

Comment:
This must be require, as if it is empty you will be reading though a potentially null pointer. Use Require when you check to see if you are allowed to do something, use check to see if something is set to a value.

Reply:
Please fix this, I don't know why you changed it, it was right before.